postgres/sqlc/schema.sql

```diff
@@ -0,0 +1,166 @@
+-- This file is used by SQLC to autogenerate the Go code needed by the database driver. 
+-- It contains all tables required for storing user data needed by the Gateway.
+-- See: https://docs.sqlc.dev/en/latest/tutorials/getting-started-postgresql.html#schema-and-queries
+
+-- It uses the tables defined in the Grove Portal database schema defined in the Portal HTTP DB (PHD) repo:
+-- https://github.com/pokt-foundation/portal-http-db/blob/master/postgres-driver/sqlc/schema.sql
+
+-- The `portal_applications` and its associated tables are converted to the `proto.GatewayEndpoint` format.
+
+-- Plans Tables
+CREATE TABLE pay_plans (
+    plan_type VARCHAR(25) PRIMARY KEY,
+    monthly_relay_limit INT NOT NULL, -- GatewayEndpoint.RateLimiting.ThroughputLimit
+    throughput_limit INT NOT NULL -- GatewayEndpoint.RateLimiting.CapacityLimit
+);
+
+-- Accounts Tables
+CREATE TABLE accounts (
+    id VARCHAR(10) PRIMARY KEY, -- GatewayEndpoint.UserAccount.AccountId
+    plan_type VARCHAR(25) NOT NULL REFERENCES pay_plans(plan_type) -- GatewayEndpoint.UserAccount.PlanType
+);
+
+-- Users Tables
+CREATE TABLE users (
+    id VARCHAR(10) PRIMARY KEY
+);
+
+-- User Auth Providers Tables
+CREATE TYPE auth_type AS ENUM ('auth0_github', 'auth0_username', 'auth0_google');
+CREATE TABLE user_auth_providers (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR(10) NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    type auth_type NOT NULL,
+    provider_user_id VARCHAR(255) NOT NULL, -- GatewayEndpoint.Auth.AuthorizedUsers[key]
+    UNIQUE(user_id, type)
+);
+
+-- Account Users Tables
+CREATE TABLE account_users (
+    id SERIAL PRIMARY KEY,
```

Comment:
Why not use (user_id, account_id) as a composite key for this table?

Reply:
This is how the Portal DB does it. It's not this repo's responsibility to make any changes to the existing database schema; it only needs to read data from it.